spec/src/main/asciidoc/core/invokers.asciidoc

```diff
@@ -171,4 +169,36 @@ public interface InvokerBuilder<T> {
 
 Calling `InvokerBuilder.build()` produces an opaque token (`InvokerInfo`) that can be passed as a parameter to a `SyntheticBeanBuilder` or `SyntheticObserverBuilder` and materializes as an `Invoker` at application runtime.
 
-// TODO lookups, transformers, wrappers
+[[invoker_lookups]]
+==== Configuring invoker lookups
+
+The `InvokerBuilder` allows configuring that the `instance` or any of the `arguments` passed to `Invoker.invoke()` should be ignored and a value should be looked up from the CDI container instead.
+
+[source,java]
+----
+public interface InvokerBuilder<T> {
+    InvokerBuilder<T> withInstanceLookup();
+    InvokerBuilder<T> withArgumentLookup(int position);
+}
+----
+
+When `withInstanceLookup()` is called on an invoker builder and the target method is not `static`, the `invoke()` method of the built invoker shall ignore the `instance` argument and instead obtain and use a contextual reference for the target bean and the bean type that declares the target method.
+Calling `withInstanceLookup()` on an invoker builder for a `static` target method has no effect.
```

Comment:
Should it not throw an exception in this case?

This did also make me question why we would restrict invokers to only work on managed beans, but allow them to work on static methods.

Reply:
Whether `withInstanceLookup()` for a `static` method should throw an exception, I'm not sure. I guess I don't really have an opinion on that.